src/main/java/com/adaptavist/tm4j/jenkins/extensions/configuration/Tm4jGlobalConfiguration.java

```diff
@@ -46,47 +48,67 @@ public boolean configure(StaplerRequest request, JSONObject formData) throws For
         Permissions.checkAdminPermission();
         request.bindParameters(this);
         Object formJiraInstances = formData.get("jiraInstances");
+        JSONArray jiraInstancesList = new JSONArray();
+        if(formJiraInstances instanceof JSONArray){
+            jiraInstancesList = formData.getJSONArray("jiraInstances");
+        }else{
+            jiraInstancesList.add(formData.getJSONObject("jiraInstances"));
+        }
+
+        List<FormInstance> newJiraInstances = new ArrayList<>();
+        for (Object jiraInstance : jiraInstancesList.toArray()) {
+            JSONObject instance = ((JSONObject)jiraInstance).getJSONObject("type");
+            FormInstance formInstance = new FormInstance();
+            formInstance.setValue(instance.getString("value"));
+            if (formInstance.getValue().equalsIgnoreCase(CLOUD_TYPE)){
+                formInstance.setCloudAddress((String) instance.getOrDefault("cloudAddress", null));
+                formInstance.setJwt((String) instance.getOrDefault("jwt", null));
+            }else{
```

Comment:
I think I'd also compare if the value is "server" to avoid typos on users' end going through.